src/snowcli/cli/nativeapp/init.py

```diff
@@ -238,76 +223,115 @@ def _init_with_url_and_template(
                 depth=1,
             )
 
-            # Move native-apps-basic to current_working_directory and rename to name
+            template_root = (
+                temp_path if use_whole_repo_as_template else temp_path / template_name
+            )
+            if not template_root.is_dir():
+                raise TemplateNotFoundError()
+
+            if use_whole_repo_as_template:
+                # Remove all git history before we move the repo
+                rmtree(template_root.joinpath(".git").resolve())
+
+            if Path.exists(template_root / "snowflake.yml.jinja"):
+                # Render snowflake.yml file from its jinja template
+                _render_snowflake_yml(
+                    parent_to_snowflake_yml=template_root,
+                    project_identifier=project_identifier,
+                )
+
+            # If not an official Snowflake Native App template
+            if git_url != SNOWFLAKELABS_GITHUB_URL:
+                _validate_and_update_snowflake_yml(
+                    target_directory=template_root,
+                    project_identifier=project_identifier,
+                )
+
+            project_path.parent.mkdir(parents=True, exist_ok=True)
+
+            # Move the template to the specified path
             move(
-                src=current_working_directory / temp_dir / template,
-                dst=current_working_directory / project_name,
+                src=template_root,
+                dst=project_path,
             )
 
-        path_to_project = current_working_directory / project_name
-
-        if Path.exists(path_to_project / "snowflake.yml.jinja"):
-            # Render snowflake.yml file from its jinja template
-            render_snowflake_yml(parent_to_snowflake_yml=path_to_project)
-
-        # If not an official Snowflake Native App template
-        if git_url != SNOWFLAKELABS_GITHUB_URL:
-            validate_and_update_snowflake_yml(target_directory=path_to_project)
-
+    except TemplateNotFoundError:
+        raise
     except Exception as err:
         # If there was any error, validation on Project Definition file or otherwise,
         # there should not be any Native Apps Project left after this.
-        if path_to_project:
-            rmtree(path_to_project.resolve())
+        if project_path.exists():
+            rmtree(project_path.resolve())
 
         log.error(err)
         raise InitError()
 
 
+def _is_part_of_existing_project(path: Path):
```

Comment:
[low-priority] Could this be reused in DefinitionManager?

Reply:
Good callout, looks like DefinitionManager has similar logic, and we shouldn't have two variants of this search. How about as a quick follow-up to this change I replace this method to call DefinitionManager logic? At first glance, it looks like it might be worth modifying DefinitionManager to office expose so functionality to search for a project definition. It looks a bit internal right now.

Reply:
Sounds good to me!